src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -155,6 +161,10 @@ protected function execute(InputInterface $input, OutputInterface $output): void
             $receiver = new StopWhenTimeLimitIsReachedReceiver($receiver, $timeLimit, $this->logger);
         }
 
+        if ($input->getOption('silent')) {
+            $receiver = new FailSilentlyReceiver($receiver);
```

Comment:
What about using DI for this instead of hardcoding ? It will allow to add parameters to your decorator (see https://github.com/symfony/symfony/pull/29132/files#r232573819) or to customize totally your option behavior with another decorator.

Reply:
Sorry I don't understand how you see your suggestion, could you give me an example?

Reply:
Hum... I didn't see it was done in this way for the other options. Sorry for the noise :)